lld/COFF/Writer.cpp

```diff
@@ -1035,6 +1036,17 @@ void Writer::sortSections() {
       sortBySectionOrder(it.second->chunks);
 }
 
+void Writer::calculateStubDependentSizes() {
+  if (ctx.config.dosStub)
+    dosStubSize = ctx.config.dosStub->getBufferSize();
+  else
+    dosStubSize = sizeof(dos_header) + sizeof(dosProgram);
+
+  coffHeaderOffset = dosStubSize + sizeof(PEMagic);
```

Comment:
Since the PEMagic needs to be 8-byte aligned, I think we need to round up dosStubSize here: `alignTo(dosStubSize, 8)`.

Also, we need to audit all the uses of dosStubSize. For example, `Writer::writePEChecksum()` uses dosStubSize to locate the coffHeader. It should use coffHeaderOffset instead.